Feat/#1197 knowledge base editor

AndrewDavydenko opened the pull request:
added new logic to sidemenu
new view of sidemenu
added nb icons for sidemenu
new privacy options - in progress
icon choosing, editing articles - in progress

-   [x] Have you followed the [contributing guidelines](https://github.com/ever-co/gauzy/blob/master/.github/CONTRIBUTING.md)?
-   [x] Have you explained what your changes do, and why they add value?

**Please note: we will close your PR without comment if you do not check the boxes above and provide ALL requested information.**

---

AndrewDavydenko commented:
https://www.loom.com/share/893399786d204a228d3e91e6dc00e7c9
added new tree model
added new privacy options
added icon choosing, editing articles
custom drag&drop - in progress
database and seeding - in progress
also have 1 issue with privacy(asked Rachit Magon in slack for it)

evereq commented:
@AndrewDavydenko it's better but still you have very serious issues with UI / UX on this page... 
For example:
1) if you have that feature to make private / public, why you need Button for that? You can just click on that eye icon and toggle private / public. 
2) when you select some element in the tree, it's not clear which one you selected. So next when you click some button it's not clear what element (leaf) in the tree it apply to.
3) to change icon for leaf in the tree if it's easy to just click on such icon, display some tiny popup with ability to select inside of that popup any icon. For the first time, when you want to add icon to leaf, you can do it this way: user select leaf, and above the tree you show very small button "add icon". If user click on it, you display same popup as for change of icon.
4) All functionality related to tree should be above tree itself (like in 3), NOT where you put it for "Article" now
5) Features for Articles should be above article text
and so on.

I think it will be best if you take say 1 hour and speak carefully with Olha about it. She did a great design for Employees Candidates and other features, I am sure she can help you. Otherwise, you are wasting days on changes which don't make it much better...

evereq commented on angular.json:
```diff
@@ -1,453 +1,437 @@
-  "defaultProject": "gauzy"
-}
\ No newline at end of file
+	"$schema": "./node_modules/@angular/cli/lib/config/schema.json",
```
@AndrewDavydenko what did you changed in `anglular.json` file and why? If nothing why you pushed it? Can you please check **every** file **before** you push it, even if it's work in progress? Code which you pushed to Github should be clean and only with changes you need

AndrewDavydenko replied:
I checked it, I discarded those changes, it updated automatically, sorry for that, I'll be more careful

evereq commented on libs/models/src/lib/help-center-menu.model.ts:
```diff
@@ -1,9 +1,70 @@
 	children?: IHelpCenter[];
 }
+
+export const helpCenterMenuList: IHelpCenter[] = [
```
@AndrewDavydenko why we have this sh..t again? It should be in seeds or not be at all, but clearly not in the model.ts file!

AndrewDavydenko replied:
I updated the structure of the tree generally because I talked to rmagon and  decided to change this to avoid minor problems, so I returned it to update front part

AndrewDavydenko replied:
@evereq  you can see the video with UI, the build is failing because I've already started the work with API.
https://www.loom.com/share/dc12bf41eb354494849461855e7e7221

AndrewDavydenko commented:
https://www.loom.com/share/dc12bf41eb354494849461855e7e7221

added modal for icons
added UI features 
API part now in work(not ready for a push at the moment)

evereq commented:
@AndrewDavydenko ok, looks much better now, thank you! Let me know when you push backend so we can merge that.

AndrewDavydenko commented:
if checks will be ok, this PR will ready for push
added API part(adding, editing, removing)
added seeding with the first layer
in progress:
db with recursive,
drag&drop,
saving drag&drop in db

evereq commented:
@AndrewDavydenko merged with "Squash", so please continue on some new feature branch after you pull the latest from `develop`. Also, please create a more "realistic" seed for initial data for testing. For example, you can just create few basic Knowledgebases for our platforms, like "Gauzy Platform" and "Ever Platform", add a few small articles to each of them, some icons, etc. So when we run app, we see some data close to real.
